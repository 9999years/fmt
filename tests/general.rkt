#lang racket

(abc ; def
)

(; def
 abc)

(abc
 def ; ghi
    )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;


'(abc ; def
 )

'(; def
 abc)

'(abc
 def ; ghi
 )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

#;(abc ; def
  )

#;(; def
  abc)

#;(abc
  def ; ghi
  )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

#; ; abc
def

' ; abc
def

#; 'abc

'#;abc
def

#;#;#;
a
b
c



' ; abc
' ; def
g


(let loop)


(foo 'abc

     def)

(match #;(x) ; def
  y
  [z 1])

#;(aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
#;(aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)

(for/list ([x 10]) #:break (even? x) 1)

(syntax-parse stx #:disable-colon-notation #:context 1 #:literals #;123 2 #:track-literals
              [hello #:with a 3 #:when bbb #:fail-when blah "def" world])


(for ([a (in-list xs)]
      #:unless
      ;; a comment is here
      (equal? a 1))
  a)

(for ([a (in-list xs)]
      #:unless
      #;(a comment is here)
      (equal? a 1))
  a)

(syntax-parse stx
  [b
   #:with b #:with a 1 #:with #:with z 3
   2])
