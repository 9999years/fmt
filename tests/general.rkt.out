#lang racket

(abc ; def
 )

; def
(abc)

(abc def ; ghi
     )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

'(abc ; def
  )

; def
'(abc)

'(abc def ; ghi
      )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

#;(abc ; def
   )

; def
#;(abc)

#;(abc def ; ghi
       )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

#;
; abc
def

; abc
'def

#;'abc

#;abc
'def

#;#;#;
a
b
c

; abc
; def
''g

(let loop
  )

(foo 'abc
     
     def)